DICOM/src/Microsoft.Health.DeID.SharedLib/DateShiftFunction.cs

```diff
@@ -0,0 +1,77 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Text;
+using EnsureThat;
+using Microsoft.Health.Dicom.DeID.SharedLib.Exceptions;
+using Microsoft.Health.Dicom.DeID.SharedLib.Settings;
+
+namespace Microsoft.Health.Dicom.DeID.SharedLib
+{
+    public class DateShiftFunction
+    {
+        public DateShiftFunction(DateShiftSetting dateShiftSetting = null)
+        {
+            dateShiftSetting ??= new DateShiftSetting();
+            DateShiftKey = dateShiftSetting.DateShiftKey;
+            DateShiftKeyPrefix = dateShiftSetting.DateShiftKeyPrefix;
+            DateShiftRange = dateShiftSetting.DateShiftRange;
+        }
+
+        public string DateShiftKey { get; set; }
+
+        public string DateShiftKeyPrefix { get; set; }
+
+        public uint DateShiftRange { get; set; }
+
+        public string ShiftDate(string inputString, string inputDateFormat = null, string outputDateFormat = null, IFormatProvider provider = null)
+        {
+            EnsureArg.IsNotNull(inputString, nameof(inputString));
+
+            DateTimeOffset date = DateTimeUtility.ParseDateTimeString(inputString, inputDateFormat, provider, DeIDGlobalSettings.DateFormat);
+
+            return ShiftDateTimeOffset(date).ToString(outputDateFormat ?? DeIDGlobalSettings.DateFormat);
+        }
+
+        public string ShiftDateTime(string inputString, string inputDateTimeFormat = null, string outputDateTimeFormat = null, IFormatProvider provider = null)
+        {
+            EnsureArg.IsNotNull(inputString, nameof(inputString));
+
+            DateTimeOffset date = DateTimeUtility.ParseDateTimeString(inputString, inputDateTimeFormat, provider, DeIDGlobalSettings.DateTimeFormat);
+
+            return ShiftDateTimeOffset(date).ToString(outputDateTimeFormat ?? DeIDGlobalSettings.DateTimeFormat);
+        }
+
+        public DateTimeOffset ShiftDateTimeOffset(DateTimeOffset dateTime)
+        {
+            EnsureArg.IsNotNull<DateTimeOffset>(dateTime, nameof(dateTime));
+
+            try
+            {
+                DateTimeOffset newDateTime = new DateTimeOffset(dateTime.Year, dateTime.Month, dateTime.Day, 0, 0, 0, dateTime.Offset);
```

Comment:
❓ **[Question]**
A little confused here.
Why do we need to create a new instance first?
Just to move clear the hour/minute/second fields, or has some other purpose?

Reply:
Just to clear the hour/minute/second fields.